src/pages/BlogPage/index.tsx

```diff
@@ -0,0 +1,266 @@
+import React, { useEffect, useState } from "react";
+import { useTranslation } from 'react-i18next';
+import useStyles from "./style";
+import useQuery from "../../hooks/useQuery";
+import { Grid, Typography, Link, Button, Breadcrumbs } from "@material-ui/core";
+import Footer from "../../components/Footer";
+import ReactMarkdown from "react-markdown";
+import { readingTime } from "../../utils/readingTime";
+import { useViewport } from "../../hooks/viewportWidth";
+import { VIEW_PORT } from "../../constants";
+import BlogsSlider from "../../components/BlogsSlider";
+import Newsletter from "../../components/Newsletter";
+
+const BlogPage: React.FC = () => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const [metadata, setMetadata] = useState<any>();
+  const [content, setContent] = useState<string>("");
+  const [previousBlog, setPreviousBlog] = useState<any>();
+  const [nextBlog, setNextBlog] = useState<any>();
+  const [currentBlogDetails, setCurrentBlogDetails] = useState<any>();
+  const [recommendedBlogs, setRecommendedBlogs] = useState<any[]>([]);
+  const queryBlogName = useQuery();
+  const axios = require('axios');
+  const parseMD = require('parse-md').default
+  const { width } = useViewport();
+  const mobileBreakpoint = VIEW_PORT.MOBILE_BREAKPOINT;
+  
+  let currentLocation = window.location.href;
+  
+  useEffect(() => {
+    const getBlogsData= async ()=>{
+      const {default: posts_json} = await import(`../../posts.json`); 
+      const currentBlog = posts_json?.filter((blog: { slug: string; }) => blog?.slug === queryBlogName)[0];
+      setPreviousBlog(posts_json?.filter((blog: { id: number; }) => blog?.id === currentBlog?.id-1)[0]);
+      setNextBlog(posts_json?.filter((blog: { id: number; }) => blog?.id === currentBlog?.id+1)[0]);
+      setCurrentBlogDetails(currentBlog);
+    }
+  
+    const fetchBlogContent=async()=>{
+      const {default: URL} = await import(`../../blogs/${queryBlogName}.md`)
+      axios.get(URL)
```

Comment:
Should we try to get the data through fetch api which is native.

Reply:
Removed it. Using posts.json